examples/dispatch.nim

```diff
@@ -0,0 +1,68 @@
+let doc = """
+Naval Fate.
+
+Usage:
+  naval_fate ship new <name>...
+  naval_fate ship <name> move <x> <y> [--speed=<kn>]
+  naval_fate ship shoot <x> <y>
+  naval_fate mine (set|remove) <x> <y> [--moored | --drifting]
+  naval_fate (-h | --help)
+  naval_fate --version
+
+Options:
+  -h --help     Show this screen.
+  --version     Show version.
+  --speed=<kn>  Speed in knots [default: 10].
+  --moored      Moored (anchored) mine.
+  --drifting    Drifting mine.
+"""
+
+import strutils
+import docopt
+import docopt/dispatch
+import sequtils
+
+let args = docopt(doc, version = "Naval Fate 2.0")
+
+# Define procedures with parameters named the same as the arguments
+proc newShip(name: seq[string]) =
+  for ship in name:
+    echo "Creating ship $#" % ship
+
+proc moveShip(name: string, x, y: int, speed: int) =
+  echo "Moving ship $# to ($#, $#) at $# kn".format(
+    name, x, y, speed)
+
+proc shootShip(x, y: int) =
+  echo "Shooting ship at ($#, $#)".format(x, y)
+
+proc setMine(x, y: int, moored, drifting: bool) =
+  echo "Setting $# mine at ($#, $#)".format(
+    (if moored: "moored" elif drifting: "drifting" else: ""),
+    x, y)
+
+proc removeMine(x, y: int) =
+  echo "Removing mine at ($#, $#)".format(x, y)
+
+# This procedure is also named the same as an argument so can be used directly
+proc mine(x, y: int, moored, remove: bool) =
+  if remove:
+    echo "Removing mine at ($#, $#)".format(x, y)
+  else:
+    echo "Setting $# mine at ($#, $#)".format(
+      (if moored: "moored" else: "drifting"),
+      x, y)
+
+if [
+  args.dispatchProc(newShip, "ship", "new"), # Runs newShip when "ship" and "new" is set
+  args.dispatchProc(moveShip, "ship", "move"), # Runs newShip when "ship" and "move" is set
+  args.dispatchProc(shootShip, "ship", "shoot"), # Runs newShip when "ship" and "shoot" is set
+  args.dispatchProc(mine) # Runs mine when "mine" is set
+].anyIt(it == true): # If any of the above returned true
```

Comment:
Writing the example like this calls all of the dispatches always, even if one of them matched. It also doesn't preclude the possibility that more than one of them matched. Is that good? Wouldn't you be better served by a short-circuiting `or` operator?

Reply:
For an example this seems fine to me. As the message in the next line clearly says the intent is simply to check whether anything ran in the first place. 
In practice a user will surely decide how to handle dispatching the procs.

Reply:
Because of the way docopt works it shouldn't be possible to get more than one of those to match, so only one of them would run. But you're right that doing it this way would conceptually mean that all of the matching procedures would be run, which may or may not be what you wanted to happen.